diesel_cli/src/print_schema.rs

```diff
@@ -202,13 +202,21 @@ pub fn output_schema(
 
         let mut all_types = table_data
             .iter()
-            .flat_map(|t| t.column_data.iter().map(|c| &c.ty))
-            .filter(|t| !diesel_provided_types.contains(&t.rust_name as &str))
-            .cloned()
+            .flat_map(|t| {
+                t.column_data.iter().map(move |c| CustomColumnType {
+                    table_name: t.name.clone(),
+                    column_name: c.rust_name.clone(),
+                    column_type: c.ty.clone(),
+                })
+            })
+            .filter(|t| !diesel_provided_types.contains(&t.column_type.rust_name as &str))
             .collect::<Vec<_>>();
 
-        all_types.sort_unstable_by(|a, b| a.rust_name.cmp(&b.rust_name));
-        all_types.dedup_by(|a, b| a.rust_name.eq(&b.rust_name));
+        all_types.sort_unstable_by(|a, b| a.column_type.rust_name.cmp(&b.column_type.rust_name));
+
+        #[cfg(feature = "postgres")]
```

Comment:
That should be done for all backends, as it simplifies the code and we do note expect to have duplicates for mysql

Reply:
I think at this point in the code we can only do de-duplication for custom postgres types, since they have separate definitions and unique names. For my mysql database of 30-40 enums and one set this dedup reduces them to two elements: "enum" and "set" and we lose everything else.

Reply:
That's a good point. It's also a bug in your current implementation. It's quite common that users install diesel-cli with more than one feature enabled. In that case this code will be executed for the generating a schema from a mysql database as well. This would result in the issue you've described above. You can reproduce  that by building diesel-cli with the following command: `cargo build --no-default-features --features "postgres mysql"`.

So that's something that needs to be fixed. 

Maybe by just doing something like:

```rust
if backend == Backend::Pg {
    // reason why we only dedup for postgres
    all_types.dedup_by(|a, b| a.column_type.rust_name.eq(&b.column_type.rust_name));
}
```